client/setup/setup-site/design/index.tsx

```diff
@@ -0,0 +1,105 @@
+import DesignPicker, { useCategorization } from '@automattic/design-picker';
+import { englishLocales } from '@automattic/i18n-utils';
+import { shuffle } from '@automattic/js-utils';
+import { useTranslate } from 'i18n-calypso';
+import { useEffect, useMemo } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import FormattedHeader from 'calypso/components/formatted-header';
+import { getRecommendedThemes as fetchRecommendedThemes } from 'calypso/state/themes/actions';
+import { getRecommendedThemes } from 'calypso/state/themes/selectors';
+import { BackButton } from '../components/back-button';
+
+// Ideally this data should come from the themes API, maybe by a tag that's applied to
+// themes? e.g. `link-in-bio` or `no-fold`
+const STATIC_PREVIEWS = [ 'bantry', 'sigler', 'miller', 'pollard', 'paxton', 'jones', 'baker' ];
+
+export function Design(): React.ReactElement {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const apiThemes = useSelector( ( state ) =>
+		getRecommendedThemes( state, 'auto-loading-homepage' )
+	);
+
+	useEffect(
+		() => {
+			if ( ! apiThemes.length ) {
+				dispatch( fetchRecommendedThemes( 'auto-loading-homepage' ) );
+			}
+		},
+		// Ignoring dependencies because we only want these actions to run on first mount
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+		[]
+	);
+
+	const designs = useMemo( () => {
+		// TODO fetching and filtering code should be pulled to a shared place that's usable by both
+		// `/start` and `/new` onboarding flows. Or perhaps fetching should be done within the <DesignPicker>
+		// component itself. The `/new` environment needs helpers for making authenticated requests to
+		// the theme API before we can do this.
```

Comment:
It might make sense to just _temporarily_ pull in the code that does authenticated requests from Calypso.

Reply:
> It might make sense to just temporarily pull in the code that does authenticated requests from Calypso.

Just chatting with @tyxla and he's optimistic with the progress being made to package-ify `lib/wp`, so optimistically pulling it in ahead of that should be fine I think.

> Aside; why theme fetching API does need authentication? It shouldn't have anything specific to the customer? I understand the argument for actually applying the design, of course.

Maybe I was wrong about this ... when I go to the gallery at `https://wordpress.com/themes` in incognito it makes a request to the same API endpoint. I'll check this in the morning, maybe it doesn't need to be authenticated after all. I thought it might have been there to show special themes to a12s only or something like that 🤷

Reply:
The themes API can likely show you themes you've uploaded, or premium themes only when you're eligible. But that's no concern for new customer onboarding in short term.